akka-remote/src/main/scala/akka/remote/security/provider/DelegatingKeyManagerFactory.scala

```diff
@@ -0,0 +1,47 @@
+package akka.remote.security.provider
```

Comment:
As all these are public API we must take extra care to make things final when possible and think carefully about the case class usage (as each case class get's copy/apply/unapply which must be kept forever because of binary compatibility), so please take an extra look with that in mind.

I think it would also be very good with a short scaladoc on each which says in a concise way what they allow you to do.

Reply:
Actually I don't think these need to be public - the factory methods on `TrustManagerFactorySetup` are sufficient I think.